### PR TITLE
Membership services: small refactor

### PR DESCRIPTION
**Pull request by millerf**
## Proposed changes:

* Remove unused class Subscription_Service
* Rename files for better reading
* Use newly added Asbtract_Token_Subscription_Service::BLOG_SUB_PENDING
 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* No functionnality change

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-asbtract-token-subscription-service.php [hunk 1]
```diff
@@ -16,7 +16,7 @@
  *
  * @package Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service
  */
-abstract class Token_Subscription_Service implements Subscription_Service {
+abstract class Asbtract_Token_Subscription_Service implements Subscription_Service {
```
Comment (lezama) [inline projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-asbtract-token-subscription-service.php]:
```suggestion
abstract class Abstract_Token_Subscription_Service implements Subscription_Service {
```

File: projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-jetpack-token-subscription-service.php [hunk 2]
```diff
@@ -16,7 +16,7 @@
  *
  * @package Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service
  */
-class Jetpack_Token_Subscription_Service extends Token_Subscription_Service {
+class Jetpack_Token_Subscription_Service extends Asbtract_Token_Subscription_Service {
```
Comment (lezama) [inline projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-jetpack-token-subscription-service.php]:
```suggestion
class Jetpack_Token_Subscription_Service extends Abstract_Token_Subscription_Service {
```

**Comment by lezama** [comment]
makes sense! 

I think we should extract all these subscription services out to a more common place so they can be used across other features and blocks.

```projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/```

**Comment by millerf** [comment]
> makes sense!
> 
> I think we should extract all these subscription services out to a more common place so they can be used across other features and blocks.
> 
> `projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/`

I just wanted to make some quick sanity out of this. 
There is still a lot more to look into. But we can definitely move it. `projects/plugins/jetpack/extensions/shared/memberships/_inc/` for example. But I am not sure this is deployed to WPCOM... I'll look into it...

File: projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-wpcom-online-subscription-service.php [hunk 3]
```diff
@@ -65,9 +65,9 @@ public function get_subscriber_email() {
 	/**
 	 * Returns true if the current authenticated user is subscribed to the current site.
 	 *
-	 * @return boolean
+	 * @return boole
```
Comment (chrismccluskey) [inline projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-wpcom-online-subscription-service.php:68]: Misspelling: `bool`

**Review by chrismccluskey — APPROVED** [review]
LGTM. I was a little confused about removing Token_Subscription but I see it's not actually used anywhere currently.

**Comment by millerf** [comment]
> LGTM. I was a little confused about removing Token_Subscription but I see it's not actually used anywhere currently.

I know...

**Pull request merged**